Blish HUD/GameServices/Modules/Managers/Gw2ApiManager.cs

```diff
@@ -1,39 +1,53 @@
 ﻿using Gw2Sharp.WebApi;
 using System.Collections.Generic;
 using System.Linq;
+using Blish_HUD.Gw2WebApi;
+using Gw2Sharp;
 using Gw2Sharp.WebApi.V2;
 using Gw2Sharp.WebApi.V2.Models;
 
 namespace Blish_HUD.Modules.Managers {
     public class Gw2ApiManager {
 
-        private readonly HashSet<TokenPermission> _permissions;
+        private const int SUBTOKEN_LIFETIME = 7;
 
-        private Connection _gw2ApiConnection;
+        private static readonly List<Gw2ApiManager> _apiManagers = new List<Gw2ApiManager>();
 
-        private Gw2WebApiClient _gw2ApiClient;
+        internal static void RenewAllSubtokens() {
+            foreach (var apiManager in _apiManagers.ToArray()) {
```

Comment:
You shouldn't need to call `.ToArray()` here.

Reply:
Good catch.